src/Juvix/Compiler/Backend/Markdown/Data/MkJuvixBlockOptions.hs

```diff
@@ -0,0 +1,61 @@
+module Juvix.Compiler.Backend.Markdown.Data.MkJuvixBlockOptions where
+
+import Juvix.Parser.Error.Base
+import Juvix.Prelude.Base
+import Juvix.Prelude.Parsing hiding (runParser)
+import Juvix.Prelude.Path
+import Text.Megaparsec qualified as P
+import Text.Megaparsec.Char.Lexer qualified as L
+
+data MkJuvixBlockOptions
+  = MkJuvixBlockOptionsHide
+  | MkJuvixBlockOptionsShow
+  | MkJuvixBlockOptionsExtractModule JuvixBlockOptionsExtractModule
+  deriving stock (Eq, Show)
+
+newtype JuvixBlockOptionsExtractModule = JuvixBlockOptionsExtractModule
+  { _juvixBlockOptionsExtractModuleDrop :: Int
+  }
+  deriving stock (Eq, Show)
+
+makeLenses ''JuvixBlockOptionsExtractModule
+
+optionExtractModuleStatements :: (IsString s) => s
+optionExtractModuleStatements = "extract-module-statements"
+
+optionHide :: (IsString s) => s
+optionHide = "hide"
+
+parseJuvixBlockOptions :: Path Abs File -> Text -> Either MegaparsecError MkJuvixBlockOptions
+parseJuvixBlockOptions p = mapLeft MegaparsecError . P.runParser parseOptions (toFilePath p)
+
+type Parser = P.Parsec Void Text
+
+spaceConsumer :: Parser ()
+spaceConsumer = L.space space1 empty empty
+
+lexeme :: Parser a -> Parser a
+lexeme = L.lexeme spaceConsumer
+
+symbol :: Text -> Parser ()
+symbol = void . L.symbol spaceConsumer
+
+decimal :: Parser Int
+decimal = lexeme L.decimal
+
+parseOptions :: Parser MkJuvixBlockOptions
+parseOptions = spaceConsumer >> parseHide <|> parseExtractModule <|> parseShow
```

Comment:
I think with do notation is easier to read. We should remove the other `eof` occurences in this module and have it only here.
```suggestion
parseOptions = do
  spaceConsumer
  opts <- parseHide 
    <|> parseExtractModule 
    <|> parseShow
  eof
  return opts
```

Reply:
Thanks 🌟